examples/frameworks/strands_demo/bedrock_agentcore/scripts/update_nat.py

```
@@ -12,20 +12,54 @@
-    networkConfiguration={"networkMode": "PUBLIC"},
-    roleArn='<IAM_AGENTCORE_ROLE>')
+
+def get_secret(secret_name: str, region_name: str) -> dict[str, str]:
+    """Retrieve secret from AWS Secrets Manager."""
+    session = boto3.session.Session()
+    secrets_client = session.client(service_name='secretsmanager', region_name=region_name)
+
+    try:
+        get_secret_value_response = secrets_client.get_secret_value(SecretId=secret_name)
+    except Exception as e:
+        raise RuntimeError(f"Error retrieving secret: {e}") from e
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)